target/linux/generic/pending-6.6/730-net-phy-realtek-detect-early-version-of-RTL8221B.patch

```diff
@@ -10,37 +10,18 @@ Implement custom identify function using the PKGID instead of iterating
 over the implemented MMDs.
 
 Signed-off-by: Daniel Golle <daniel@makrotopia.org>
----
- drivers/net/phy/realtek.c | 50 ++++++++++++++++++++++++++++++++++++++-
- 1 file changed, 49 insertions(+), 1 deletion(-)
 
 --- a/drivers/net/phy/realtek.c
 +++ b/drivers/net/phy/realtek.c
-@@ -81,6 +81,7 @@
- 
- #define RTL_GENERIC_PHYID			0x001cc800
- #define RTL_8211FVD_PHYID			0x001cc878
-+#define RTL_8221B_VB_CG				0x001cc849
- 
- MODULE_DESCRIPTION("Realtek PHY driver");
- MODULE_AUTHOR("Johnson Leung");
-@@ -801,6 +802,54 @@ static int rtl822x_probe(struct phy_devi
- 	return 0;
+@@ -782,6 +782,38 @@ static int rtl8226_match_phy_device(stru
+ 	       rtlgen_supports_2_5gbps(phydev);
  }
  
 +static int rtl8221b_vb_cg_match_phy_device(struct phy_device *phydev)
 +{
 +	int val;
 +	u32 id;
 +
-+	if (phydev->is_c45) {
-+		if (phydev->c45_ids.device_ids[1])
-+			return phydev->c45_ids.device_ids[1] == RTL_8221B_VB_CG;
-+	} else {
-+		if (phydev->phy_id)
-+			return phydev->phy_id == RTL_8221B_VB_CG;
-+	}
```

Comment:
Could someone please check what is this value of `phydev->c45_ids.device_ids[1]` and `phydev->phy_id`. 
I assume that these values are different between kernel 6.1 and 6.6.

Reply:
I tired this with kernel 6.6 (without the patch):
```C
static int rtl8221b_vb_cg_match_phy_device(struct phy_device *phydev)
{
	int val;
	u32 id;

	if (phydev->is_c45) {
        pr_info("phydev->phy_id: %X\n", phydev->phy_id);
        pr_info("phydev->c45_ids.device_ids[1]: %X\n", phydev->c45_ids.device_ids[1]);
```

This gives me the following output:
```
Sat May 11 00:32:08 2024 kern.info kernel: [    1.315666] phydev->phy_id: 0
Sat May 11 00:32:08 2024 kern.info kernel: [    1.318635] phydev->c45_ids.device_ids[1]: FFFFFFFF
```